### DISCUSSION TITLE
Can a userscript be detected with window.onerror?

### DISCUSSION THREAD

**Discussion by SmileB4Deathz** [Tampermonkey/tampermonkey#1763] [Q&A]
If a webpage catches errors with window.onerror and sends the error to their servers, will a tampermonkey script error also be detected with this method?

**Comment by derjanb** [comment]
In general yes. Exceptions are `JavaScript` and `DOM` `@sandbox` modes in Firefox and `DOM` mode in Chrome (if enabled).

```JS
// @sandbox      JavaScript
```

https://tmnk.net/docs#meta:sandbox
